include/pybind11/detail/internals.h

```diff
@@ -121,6 +121,10 @@ struct internals {
         PYBIND11_TLS_FREE(tstate);
     }
 #endif
+
+    inline initproc default_object_init() {
```

Comment:
Note that I'm adding this function instead of adding the function pointer to internals to avoid changing the size of the structure.

Reply:
Unless I'm misremembering stuff, we have a macro for versioning the ABI of the internals structure.

Reply:
I'd vote for just including this inline, actually (i.e., `auto &internals = get_internals();` and `((PyHeapTypeObject*) internals.instance_base)->ht_type.tp_init`). Accessing this should be very exceptional, so I'd argue to make it loud and noisy in the place it is being accessed, rather than adding a smooth function to `internals`.